lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -591,6 +592,74 @@ static bool parameterTypesMatch(const ValueDecl *derivedDecl,
   return true;
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+static bool overridesDifferentiableAttribute(const ValueDecl *derivedDecl,
+                                             const ValueDecl *baseDecl) {
+  ASTContext &ctx = baseDecl->getASTContext();
+  auto &diags = ctx.Diags;
+
+  auto *derivedAFD = dyn_cast<AbstractFunctionDecl>(derivedDecl);
+  auto *baseAFD = dyn_cast<AbstractFunctionDecl>(baseDecl);
+
+  if (!derivedAFD || !baseAFD)
+    return false;
+
+  auto derivedDAs = derivedAFD->getAttrs().getAttributes<DifferentiableAttr>();
+  auto baseDAs = baseAFD->getAttrs().getAttributes<DifferentiableAttr>();
+
+  // Make sure all the differentiable attributes in `baseDecl` are
+  // also declared in `derivedDecl`.
+  for (auto baseDA : baseDAs) {
+    auto baseParameters = AutoDiffIndexSubset::get(
+        ctx, baseDA->getParameterIndices()->parameters);
+    auto defined = false;
+    for (auto derivedDA : derivedDAs) {
+      auto derivedParameters = AutoDiffIndexSubset::get(
+          ctx, derivedDA->getParameterIndices()->parameters);
+      if (baseParameters->isSubsetOf(derivedParameters)) {
+        defined = true;
```

Comment:
I think we could add a `break;` here as we can break early once we found a match.

Reply:
Good catch!